### DISCUSSION TITLE
Token Claim: Failure in transaction sent & unknown error thrown

### DISCUSSION THREAD

**Discussion by saraliyah** [phantom/docs#74] [Help & Support]
### Summary

Background 
We are a web3 NFT studio that is developing an ERC-20 token claim website https://airdrop.tidalflats.studio/ for our users. We try to integrate with Phantom wallet to unlock claiming of ERC-20 $TIDE tokens for our users but the integration does not work. 

Problem: 
1. Users are allowed to sign in with Phantom wallet but transaction cannot be successfully sent. 
2. An unknown error will be thrown. 
3. $TIDE tokens can be added to Phantom wallet in ERC-20 list but the transfer function just fails for all users. 

See video below for more information
<img width="892" alt="image" src="https://github.com/phantom/docs/assets/136977095/c55d01ef-8610-4776-a7d1-ad20904139cd">

<img width="896" alt="image" src="https://github.com/phantom/docs/assets/136977095/a865f735-8bb2-44ba-9efd-73994ba51370">

This issue is extremely urgent, please contact our team @ sarah@meta-z.io & blair@meta-z.io. Thanks! 

### Steps to Reproduce

<img width="1248" alt="image" src="https://github.com/phantom/docs/assets/136977095/72f4687f-328a-4792-be30-e2cab1ab764e">

### Is there an existing discussion for this?

- [X] I have searched the existing discussions

**Comment by adamdelphantom** [comment]
Hello @saraliyah,

Would you mind posting your transaction call snippet?

**Comment by saraliyah** [comment]
Thanks Adam for following up on this.

+BLAIR LEE ***@***.***> Can you provide the transaction call snippet
here?

Cheers
Sarah

**Comment by nulLeeKH** [comment]
I already tried several changes. So it looks a bit messy.

``` javascript
        const block = await web3.eth.getBlock("pending")
        const baseFee = Number(block.baseFeePerGas);
        const tip = 100000000;
        const max = Number(tip) + baseFee - 1;

        const param = {
            from: accounts[0],
            to: ERC20_AIRDROP_CONTRACT_ADDRESS,
            // nonce: await window.ethereum.getTransactionCount(accounts[0]),
            value: '0x0',
            gasPrice: '0x' + Math.abs(tip).toString(16),
            gasLimit: '0x' + Math.abs(83406).toString(16),
            // maxPriorityFeePerGas: '0x' + Math.abs(tip).toString(16),
            // maxFeePerGas: '0x' + Math.abs(max).toString(16),
            // gasPrice: '0x5D21DBA00',
            type: '0x2',
            data: '0x4451d89f',
            chainId: CHAIN_ID,
            // gasLimit: gas,
            // gasPrice: gasPrice,
            // data: '0x4451d89f',
            // type: '0x2',
        }
        const response = await window.ethereum.request({
            method: 'eth_sendTransaction',
            params: [param],
        });
```

I used web3.js version 4.1.0 and vanilla js.

**Reply by adamdelphantom** [reply]
Thanks for this!

I see the alert box says 'unexpected error.'  

Can you provide the error from the request call?

**Reply by MD-Dhaval-Saxena** [reply]
i have same error,Any solution error occured in mumbai , goerli network

**Comment by MD-Dhaval-Saxena** [comment]
Any update on this @adamdelphantom

**Comment by MD-Dhaval-Saxena** [comment]
Here is My Code

```
const Transfer_token = async () => {
  try {
    let _to = "receiver_address";
    let amount = 1 * 10 ** 6;

    const transfer_coin = Usdt_ABI.find((i) => i.name === "transfer");
    const interfaces = new ethers.utils.Interface([transfer_coin]);
    const encodedFunction = interfaces.encodeFunctionData(
      `${transfer_coin.name}`,
      [_to, amount]
    );
    
    const gasToSend = await getEstimateGas(
      WALLET_CONNECTED,
      Contract_List.usdt_address,
      encodedFunction
    );
    const result = await window.phantom.ethereum.request({
      method: "eth_sendTransaction",
      params: [
        {
          chainId: 5, //goerli network
          from: WALLET_CONNECTED,
          to: "Token_address", //erc20 tokenContract
          data: encodedFunction,
          gas: gasToSend.toString(),
        },
      ],
    });

    console.log(result);
  } catch (error) {
    console.log({ error });
  }
};

```

**Reply by adamdelphantom** [reply]
@MD-Dhaval-Saxena Is this related to not being able to connect to goerli or Mumbai? If so you'll need to wait for the fix mentioned in the other thread.

**Reply by MD-Dhaval-Saxena** [reply]
yes as previous thread i'm not able to do transaction

**Reply by adamdelphantom** [reply]
Can you provide the error you log to the console please?

**Reply by MD-Dhaval-Saxena** [reply]
![image](https://github.com/phantom/docs/assets/122987863/a73769ee-5406-4d2e-9f21-75f4bedfb028)

**Reply by MD-Dhaval-Saxena** [reply]
and ALSO check phantom causing error with metamask in same browser .

**Reply by adamdelphantom** [reply]
- Is your phantom extension set to the testnet mode?
- Is Phantom connected prior to attempting the transfer?
- Do you have any other wallets other than Phantom and Metamask installed on this browser session? 

It's also worth trying another browser with just Phantom installed to help isolate the issue.

**Reply by MD-Dhaval-Saxena** [reply]
-Is your phantom extension set to the testnet mode? => Yes it's set on test mode and active network is etherum goerli
-Is Phantom connected prior to attempting the transfer? =>Yes it's connected
-Do you have any other wallets other than Phantom and Metamask installed on this browser session? => No other wallet

![Screenshot (270)](https://github.com/phantom/docs/assets/122987863/f04299d8-8663-4981-a919-c67af2b3baa2)


-trying another browser with just Phantom installed to help isolate the issue.

=>i tried in chrome latest with only phantom enabled 


for helping you out i have more to share.

so i have few different errors

-this is most occurring after signing transaction

1) `Error: Unexpected error\n    at  oa.<anonymous> (<anonymous>:7:4037)\n    at Generator`


2) when i am calling transfer method it pop up the phantom screen for signing transaction but there is error that` "Transaction Error(OutOfGas(BasicOutofGas))"`
![Screenshot (271)](https://github.com/phantom/docs/assets/122987863/e3e94a1f-a4b7-47dd-b1f6-dd62bfaf5f06)


-there was a problem estimating the fee for this transaction it may fail.

note: i am sending gas calculation in my transaction 


3) sometime this error appears when i am calling  the transfer method:

 `Error: Requested resource not available.\n    at oa.<anonymous> (<anonymous>:`
 
4) from my side cross verified that i am on correct network and i have sufficient balance and tokens to transfer

**Reply by MD-Dhaval-Saxena** [reply]
@adamdelphantom Any update on this ?

**Reply by adamdelphantom** [reply]
@MD-Dhaval-Saxena Do you have this hosted anywhere I can interact with and is the code open source so I can browse?

**Comment by MD-Dhaval-Saxena** [comment]
> @MD-Dhaval-Saxena Do you have this hosted anywhere I can interact with and is the code open source so I can browse?

Here is github repo-https://github.com/MD-Dhaval-Saxena/web3phantom/tree/main

here is live site: https://papaya-sawine-534187.netlify.app/

**Reply by adamdelphantom** [reply]
@MD-Dhaval-Saxena it appears you're attempting to connect to Mumbai correct? We don't yet support Mumbai. Should be coming in the future, but I don't have a timeline to provide.
![Screenshot 2023-11-08 at 10 42 34 AM](https://github.com/phantom/docs/assets/64601408/ecef5f74-d486-4a5b-9141-684659515cfb)

**Comment by MD-Dhaval-Saxena** [comment]
> @MD-Dhaval-Saxena it appears you're attempting to connect to Mumbai correct? We don't yet support Mumbai. Should be coming in the future, but I don't have a timeline to provide. ![Screenshot 2023-11-08 at 10 42 34 AM](https://user-images.githubusercontent.com/64601408/281459818-ecef5f74-d486-4a5b-9141-684659515cfb.png)


But It's not only mumbai is not working in testnet mode Etherum goerli also there ,even goerli network is not working!

![image](https://github.com/phantom/docs/assets/122987863/f3f3d6e8-01fd-4392-8652-22bc93e953bf)

- Goerli and mumbai shown in testnet but both are not working.


Attaching Live Website and Code to review.

website:- https://papaya-sawine-534187.netlify.app/
github:-https://github.com/MD-Dhaval-Saxena/web3phantom/tree/main

**Reply by adamdelphantom** [reply]
Why are you using window.solana.connect and not window.ethereum here?

https://github.com/MD-Dhaval-Saxena/web3phantom/blob/31f012884a8301e5ab75531f012ebba3c4504ae0/public/script.js#L380

**Reply by MD-Dhaval-Saxena** [reply]
I found somewhere in community forum so i added but after your reply i tried without window.solana.connect and it's working fine for establishing connection but the issue for transfer is the same.
